### PR TITLE
Dependencies/fix versions 12 21

### PR DESCRIPTION
**Pull request by josiahhaswell**
A CVE scan yields several security vulnerabilities in transitive dependencies:

- [CVE-2020-14040](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2020-14040) (x/text)
- [CVE-2021-38561](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2021-38561) (x/text)
- [CVE-2022-32149](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2022-32149)
- [CVE-2023-39325](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2023-39325) (x/net)
- [CVE-2023-3978](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2023-3978) (x/net)
- [CVE-2024-45338](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2024-45338) (x/net)
- [CVE-2021-38561](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2021-38561) (x/text)
- [CVE-2022-32149](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2022-32149) (x/text)
- [CVE-2022-41723](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2022-41723) (x/net)
- [CVE-2023-39325](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2023-39325) (x/net)
- [CVE-2023-3978](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2023-3978) (x/net)
- [CVE-2024-45338](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2024-45338) (x/net)
- [CVE-2022-32149](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2022-32149) (x/text)

# Regression status

I've run the test-suite against MySQL 5.7, MySQL 8, and MariaDB 10.  Additionally, I've reviewed the changes in these dependencies between the relevant versions and do not see any impacting regressions

### CODE REVIEW

**Comment by evanelias** [comment]
What scanner did you run? I don't see any evidence of `golang.org/x/net` or `golang.org/x/text` actually being transitive dependencies of Skeema in the first place:

```
~/src/github.com/skeema/skeema $ go mod why golang.org/x/text    
# golang.org/x/text
(main module does not need package golang.org/x/text)

~/src/github.com/skeema/skeema $ go mod why -m golang.org/x/text
# golang.org/x/text
(main module does not need module golang.org/x/text)

~/src/github.com/skeema/skeema $ go mod why golang.org/x/net    
# golang.org/x/net
(main module does not need package golang.org/x/net)

~/src/github.com/skeema/skeema $ go mod why -m golang.org/x/net
# golang.org/x/net
(main module does not need module golang.org/x/net)
```

Nor do I see them when grepping the codebase or vendored code.

As far as I can see, they're only mentioned in go.sum, which can happen for a variety of reasons unrelated to being actual dependencies. In this case I suspect they're coming from the go.sum of github.com/mattn/goveralls verbatim since that contains these same lines... and they're possibly only there because that module has a `go 1.13` line, which is too old for [module graph pruning](https://go.dev/ref/mod#graph-pruning) to be enabled, if I understand correctly. In any case, Skeema doesn't even import github.com/mattn/goveralls directly; it's only used in the Github Actions CI flow.

TL;DR I don't see any relevant vulnerability here, so closing this, but if I missed something in my analysis please let me know and I can reopen.

In the future, please always discuss things on an Issue first before opening a PR, as requested by the PR template text as well as CONTRIBUTING.md. Thanks.

**Comment by evanelias** [comment]
Just to be safe, I've done the following:

* Ran [`govulncheck`](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) by hand, which confirms no known vulnerabilities in Skeema's dependencies
* Added `govulncheck` to our GitHub Actions configuration, so that CI will fail if any new vulnerabilities come up
* De-vendored `goveralls` to avoid any future false positives based on its old non-pruned go.sum entries. (`goveralls` is still used in our GitHub Actions configuration, it's just re-downloaded now at the same pinned version, instead of being vendored inside the repo.)

**Comment by josiahhaswell** [comment]
Thank you, @evanelias -- really appreciate it!  Really great project
